core/src/test/java/org/infinispan/commands/PutMapCommandStressTest.java

```diff
@@ -0,0 +1,212 @@
+      // Amend first so we can increase the transport thread pool
+      TestCacheManagerFactory.amendGlobalConfiguration(gcb, flags);
+      // we need to increase the transport and remote thread pools to default values
+      BlockingThreadPoolExecutorFactory executorFactory = new BlockingThreadPoolExecutorFactory(
+            25, 25, 10000, 30000);
+      gcb.transport().transportThreadPool().threadPoolFactory(executorFactory);
+
+      gcb.transport().remoteCommandThreadPool().threadPoolFactory(executorFactory);
+
+      EmbeddedCacheManager cm = TestCacheManagerFactory.newDefaultCacheManager(true, gcb,
+            new ConfigurationBuilder(), false);
+      cacheManagers.add(cm);
+      return cm;
+   }
+
+   public void testStressNodesLeavingWhileMultipleIterators() throws InterruptedException, ExecutionException, TimeoutException {
+      final Map<Integer, Integer> masterValues = new HashMap<Integer, Integer>();
+      final Map<Integer, Integer>[] keys = new Map[THREAD_WORKER_COUNT];
+      for (int i = 0; i < keys.length; ++i) {
+         keys[i] = new HashMap<>();
+      }
+      // First populate our caches
+      for (int i = 0; i < CACHE_ENTRY_COUNT; ++i) {
+         masterValues.put(i, i);
+         keys[i % THREAD_WORKER_COUNT].put(i, i);
+      }
+
+      cache(0, CACHE_NAME).putAll(masterValues);
+
+      for (int i = 0; i < keys.length; ++i) {
+         keys[i] = Collections.unmodifiableMap(keys[i]);
+      }
+
+      final AtomicBoolean complete = new AtomicBoolean(false);
+      final Exchanger<Throwable> exchanger = new Exchanger<Throwable>();
+      // Now we spawn off CACHE_COUNT of threads.  All but one will constantly calling to iterator while another
+      // will constantly be killing and adding new caches
+      Future<Void>[] futures = new Future[THREAD_WORKER_COUNT + 1];
+      for (int j = 0; j < THREAD_MULTIPLIER; ++j) {
+      // We iterate over all but the last cache since we kill it constantly
+      for (int i = 0; i < CACHE_COUNT - 1; ++i) {
+         final int offset = j * (CACHE_COUNT -1) + i;
+         final Cache<Integer, Integer> cache = cache(i, CACHE_NAME);
+         futures[i + j * (CACHE_COUNT -1)] = fork(new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+               Map<Integer, Integer> keysToUse = keys[offset];
+               try {
+                  int iteration = 0;
+                  
+                  while (!complete.get()) {
+                     log.tracef("Starting iteration %s", iteration);
+                     // UNCOMMENT following to test insertions by themselves
+//                     for (Entry<Integer, Integer> entry : keysToUse.entrySet()) {
+//                        cache.put(entry.getKey(), entry.getValue());
+//                     }
+                     cache.getAdvancedCache().putAll(keysToUse);
+                     // UNCOMMENT following to make sure puts are propagated properly
```

Comment:
I was doing that as to not slow down the test so it wasn't polluting the numbers.  This way I can compare the old and new implementation.